pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -67,13 +70,15 @@ func readyCondition(tpy string) runtimev1alpha1.Condition {
 }
 
 type appHandler struct {
-	r             *Reconciler
-	app           *v1beta1.Application
-	appfile       *appfile.Appfile
-	logger        logr.Logger
-	inplace       bool
-	isNewRevision bool
-	revisionHash  string
+	r               *Reconciler
+	app             *v1beta1.Application
+	appfile         *appfile.Appfile
+	logger          logr.Logger
+	inplace         bool
+	isNewRevision   bool
+	revisionHash    string
+	crossNsResource []v1beta1.TypedReference
```

Comment:
you should compare with the real status of Application object instead of use the variable in appHandler.